src/Temporalio/Client/WorkerTaskReachability.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace Temporalio.Client
+{
+    /// <summary>
+    /// Contains information about the reachability of some Build IDs.
+    /// </summary>
+    /// <param name="BuildIdReachability">Maps Build IDs to information about their reachability.</param>
+    public record WorkerTaskReachability(IReadOnlyDictionary<string, BuildIdReachability> BuildIdReachability)
+    {
+        /// <summary>
+        /// Convert from proto.
+        /// </summary>
+        /// <param name="proto">Proto.</param>
+        /// <returns>Converted value.</returns>
+        internal static WorkerTaskReachability FromProto(Api.WorkflowService.V1.GetWorkerTaskReachabilityResponse proto)
+        {
+            var idMap = proto.BuildIdReachability.ToDictionary(
+                bid => bid.BuildId,
+                bid =>
+                {
+                    var unretrieved = new List<string>();
+                    var tqrDict = bid.TaskQueueReachability
+                        .SkipWhile(tqr =>
+                        {
+                            if (tqr.Reachability.Count == 1 &&
+                                tqr.Reachability[0] == Api.Enums.V1.TaskReachability.Unspecified)
+                            {
+                                unretrieved.Add(tqr.TaskQueue);
+                                return true;
+                            }
+
+                            return false;
+                        })
+                        .ToDictionary(
+                            tqr => tqr.TaskQueue,
+                            tqr =>
+                                tqr.Reachability.Select(r => TaskReachabilityType.FromProto(r)).ToList().AsReadOnly());
+                    return new BuildIdReachability(tqrDict, unretrieved.AsReadOnly());
+                });
+            return new(idMap);
+        }
+    }
+
+    /// <summary>
+    /// Contains information about the reachability of a specific Build ID.
+    /// </summary>
+    /// <param name="TaskQueueReachability">Maps Task Queue names to the reachability status of the Build ID on that
+    /// queue. If the value is an empty list, the Build ID is not reachable on that queue.</param>
+    /// <param name="UnretrievedTaskQueues">If any Task Queues could not be retrieved because the server limits the
+    /// number that can be queried at once, they will be listed here.</param>
+    public record BuildIdReachability(
```

Comment:
I know it's a bit redundant wrt file count, but I think we want each namespace-level type to have its own file